docs/reference/data-streams/data-streams-overview.asciidoc

```diff
@@ -0,0 +1,118 @@
+[[data-streams-overview]]
+== Data streams overview
+++++
+<titleabbrev>Overview</titleabbrev>
+++++
+
+A data stream consists of one or more _backing indices_. Backing indices are
+<<index-hidden,hidden>>, automatically-generated indices used to store a
+stream's documents.
+
+image::images/data-streams/data-streams-diagram.svg[align="center"]
+
+The creation of a data stream requires an associated
+<<indices-templates,composable template>>. This template acts as a blueprint for
+the stream's backing indices. It contains:
+
+* A name or wildcard (`*`) pattern for the data stream.
+
+* The data stream's _timestamp field_. This field must be mapped as a
+  <<date,`date`>> or <<date_nanos,`date_nanos`>> field datatype and be
+  included in every document indexed to the data stream.
+
+* The mappings and settings applied to each backing index when it's created.
+
+The same composable template can be used to create multiple data streams.
+See <<set-up-a-data-stream>>.
+
+[discrete]
+[[data-streams-generation]]
+=== Generation
+
+Each data stream tracks its _generation_: a six-digit, zero-padded integer
+that acts as a cumulative count of the data stream's backing indices. This count
+includes any deleted indices for the stream. The generation is incremented
+whenever a new backing index is added to the stream.
+
+When a backing index is created, the index is named using the following
+convention:
+
+[source,text]
+----
+.ds-<data-stream>-<generation>
+----
+
+.*Example*
+[%collapsible]
+====
+The `web_server_logs` data stream has a generation of `34`. The most recently
+created backing index for this data stream is named
+`.ds-web_server_logs-000034`.
+====
+
+Because the generation increments with each new backing index, backing indices
+with a higher generation contain more recent data. Backing indices with a lower
+generation contain older data.
+
+A backing index's name can change after its creation due to a
+<<indices-shrink-index,shrink>>, <<snapshots-restore-snapshot,restore>>, or
+other operations.
+
+[discrete]
+[[data-stream-write-index]]
+=== Write index
+
+When a read request is sent to a data stream, it routes the request to all its
+backing indices. For example, a search request sent to a data stream would query
+all its backing indices.
+
+image::images/data-streams/data-streams-search-request.svg[align="center"]
+
+However, the most recently created backing index is the data stream’s only
+_write index_. The data stream routes all indexing requests to this index.
+
+image::images/data-streams/data-streams-index-request.svg[align="center"]
+
+You cannot add new documents to a stream's other backing indices, even
+by sending indexing requests directly to the index.
+
+Because it's the only index capable of handling indexing requests, you also
+cannot <<indices-close,close>> a data stream's write index.
```

Comment:
We currently prohibit closing and deleting the write index and are going to prohibit freezing, shrinking, and any other operation that would prevent ingestion into the write index even if, as in the case of shrinking, ingestion would be prevented only temporarily.

Reply:
Thanks. I've expanded the list to explicitly include close, freeze, and shrink. I also added a high-level note stating that operations that hinder ingest are prohibited.